src/singletons.js

```diff
@@ -31,7 +23,38 @@ async function ERC1820Registry (funder) {
 
   await web3.eth.sendSignedTransaction(ERC1820_REGISTRY_DEPLOY_TX);
 
-  return ERC1820RegistryArtifact.at(ERC1820_REGISTRY_ADDRESS);
+  return getDeployedERC1820Registry();
+}
+
+async function getDeployedERC1820Registry () {
+  const environment = require('./config/environment').getEnviroment();
+
+  if (environment === 'truffle') {
+    let truffleContract;
+    [ '@truffle/contract', 'truffle-contract' ].forEach(pkg => {
+      if (truffleContract === undefined) {
+        truffleContract = tryRequire(pkg);
+      }
+    });
+
+    if (truffleContract === undefined) {
+      throw new Error(`
+Current environment is 'truffle', but found no truffle contract abstraction package.
+Install it via:
+  npm install @truffle/contract`);
+    }
+
+    const contractAbstraction = truffleContract({ abi: ERC1820_REGISTRY_ABI });
+    contractAbstraction.setProvider(web3.currentProvider);
+
+    return contractAbstraction.at(ERC1820_REGISTRY_ADDRESS);
+
+  } else if (environment === 'web3') {
+    return new web3.eth.Contract(ERC1820_REGISTRY_ABI, ERC1820_REGISTRY_ADDRESS);
+
+  } else {
+    throw new Error(`Unknown runtime environment: '${environment}'`);
```

Comment:
"Runtime environment" is quite confusing IMO.

```suggestion
    throw new Error(`Unknown environment: '${environment}'`);
```